docs/UIO.md

```diff
@@ -0,0 +1,172 @@
+ultra-large-scale data (egs: more than 50,000 hours or 50 million or more audio), this method will cause the training
+to fail to run for two reasons:
+- Out of memory(OOM): The physical memory of the general machine is difficult to load the training data at one time.
+- Slow down reading performance:  In the case that the large-scale data memory cannot be used as a file cache, the training
+data reading speed is greatly reduced.
+
+## System design of UIO
+### Overall design
+Inspired by the following industrial methods(egs: [webdataset](https://github.com/webdataset/webdataset), [TFRecord](https://www.tensorflow.org/tutorials/load_data/tfrecord)),
+WeNet redesigned the IO method. Its core idea is to make the audio and labels of multiple small data(such as 1000 pieces),
+into compressed packets (tar) and read them based on the IterableDataset of Pytorch. The advantages of this method is：
+- Only the index information of the compressed package needs to be maintained in memory, which greatly saves memory and
+solves the problem of OOM.
+- The on-the-fly decompression is performed in the memory, and the data in the same compressed package is read in
+sequence, which solves the problem of slow random reading performance. Different compressed packets can be read randomly
+to ensure the global randomness of data.
+
+The new IO method takes into account both small data sets and large data sets， and provides two data reading methods.
+We call it UIO. The overall design of UIO is shown in the figure below:
+
+![UIO System Design](./images/UIO_system.png)
+
+Some necessary explanations about the above figure:
+- Small IO(raw) supports small datasets, which we call raw mode. This mode only supports local file reading.
+The required documents must be sorted into Kaldi style file: wav.scp and text.(It's the same as before)
+- Big IO supports(shared) supports large datasets, which we call shard mode. This mode can support both local file
+reading and network cloud storage file reading. The required files must be sorted into compressed packages. Audio (wav)
+and label (txt) are stored in a single compressed package in sequence.
+
+### Chain IO
+Inspired by TFRecord chain IO, UIO also adopts chain implementation. In practical use, chain IO is more flexible,
+easier to expand and easier to debug. TFRecord IO example as follows,
+```python
+def read_dataset(filename, batch_size):
+    dataset = tf.data.TFRecordDataset(filename)
+    dataset = dataset.map(_parse_image_function, num_parallel_calls=tf.data.experimental.AUTOTUNE)
+    dataset = dataset.shuffle(500)
+    dataset = dataset.batch(batch_size, drop_remainder=True)
+    dataset = dataset.repeat()
+    dataset = dataset.prefetch(buffer_size=tf.data.experimental.AUTOTUNE)
+    return dataset
+```
+Refer to TFRecord IO, the UIO dataflow in WeNet is designed as the figure below:
+
+![UIO dataflow](./images/UIO_dataflow.png)
+
```

Comment:
this image is a bit blurry （see [link](https://github.com/jinsongpan/wenet/blob/dev-pjs/docs/UIO.md#chain-io))，maybe @robin1001 has a clearer picture?

Reply:
okay, I will give @jinsongpan the clear version.

Reply:
ok, i have replace them.